propagation.go

```diff
@@ -41,71 +44,83 @@ var (
 type BuiltinFormat byte
 
 const (
-	// SplitBinary encodes the Span in a SplitBinaryCarrier instance.
+	// Binary encodes the Span in a BinaryCarrier instance.
 	//
-	// The `carrier` for injection and extraction must be a
-	// `*SplitBinaryCarrier` instance.
-	SplitBinary BuiltinFormat = iota
+	// The `carrier` for injection and extraction must be a `BinaryCarrier`
+	// instance (not a pointer to a BinaryCarrier instance).
+	Binary BuiltinFormat = iota
 
-	// SplitText encodes the Span in a SplitTextCarrier instance.
+	// TextMap encodes the Span in a TextMapCarrier instance.
 	//
-	// The `carrier` for injection and extraction must be a `*SplitTextCarrier`
-	// instance.
-	SplitText
+	// The `carrier` for injection and extraction must be a `TextMapCarrier`
+	// instance (not a pointer to a TextMapCarrier instance).
+	TextMap
 
 	// GoHTTPHeader encodes the Span into a Go http.Header instance (both the
 	// tracer state and any baggage).
 	//
 	// The `carrier` for both injection and extraction must be an http.Header
-	// instance.
+	// instance (not a pointer to an http.Header instance).
+	//
+	// If there are entries in the http.Header map prior to a call to Inject(),
+	// they are left alone (i.e., the map is not cleared). Similarly, in calls
+	// to Join() it is fine (and expected in some cases) for the http.Header
+	// map to contain other unrelated data (i.e., non-OpenTracing headers).
 	GoHTTPHeader
+
+	// SplitBinary is DEPRECATED
+	SplitBinary
+	// SplitText is DEPRECATED
+	SplitText
 )
 
-// SplitTextCarrier breaks a propagated Span into two pieces.
-//
-// The Span is separated in this way for a variety of reasons; the most
-// important is to give OpenTracing users a portable way to opt out of
-// Baggage propagation entirely if they deem it a stability risk.
+// TextMapCarrier represents a Span for propagation within a key:value map of
+// unicode strings.
+type TextMapCarrier interface {
+	// Add a key:value pair to the carrier.`key` should be prefixed in a way
+	// that protects against collisions with things like HTTP headers (which
+	// may share space with Tracer data in the TextMapCarrier).
+	Add(key, val string)
```

Comment:
@yurishkuro fair point... I guess there's no need to tell the interface implementor about this sort of thing. (Another benefit to using an `interface` rather than the raw underlying storage type)